lib/active_merchant/billing/gateways/mundipagg.rb

```diff
@@ -0,0 +1,287 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class MundipaggGateway < Gateway
+      self.live_url = 'https://api.mundipagg.com/core/v1/'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      self.homepage_url = 'https://www.mundipagg.com/'
+      self.display_name = 'Mundipagg'
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        '400' => STANDARD_ERROR_CODE[:processing_error],
+        '401' => STANDARD_ERROR_CODE[:config_error],
+        '404' => STANDARD_ERROR_CODE[:processing_error],
+        '412' => STANDARD_ERROR_CODE[:processing_error],
+        '422' => STANDARD_ERROR_CODE[:processing_error],
+        '500' => STANDARD_ERROR_CODE[:processing_error]
+      }
+
+      STANDARD_ERROR_MESSAGE_MAPPING = {
+        '400' => 'Invalid request;',
+        '401' => 'Invalid API key;',
+        '404' => 'The requested resource does not exist;',
+        '412' => 'Valid parameters but request failed;',
+        '422' => 'Invalid parameters;',
+        '500' => 'An internal error occurred;'
+      }
+
+      def initialize(options={})
+        requires!(options, :api_key)
+        super
+      end
+
+      def purchase(money, payment, options={})
+        post = {}
+        add_invoice(post, money, options)
+        add_customer_data(post, options) unless payment.is_a?(String)
+        add_shipping_address(post, options)
+        add_payment(post, payment, options)
+
+        commit('sale', post)
+      end
+
+      def authorize(money, payment, options={})
+        post = {}
+        add_invoice(post, money, options)
+        add_customer_data(post, options) unless payment.is_a?(String)
+        add_shipping_address(post, options)
+        add_payment(post, payment, options)
+        add_capture_flag(post, payment)
+        commit('authonly', post)
+      end
+
+      def capture(money, authorization, options={})
+        post = {}
+        post[:code] = authorization
+        add_invoice(post, money, options)
+        commit('capture', post, authorization)
+      end
+
+      def refund(money, authorization, options={})
+        add_invoice(post={}, money, options)
+        commit('refund', post, authorization)
+      end
+
+      def void(authorization, options={})
+        commit('void', post=nil, authorization)
+      end
+
+      def store(payment, options={})
+        post = {}
+        options.update(name: payment.name)
+        options = add_customer(options) unless options[:customer_id]
+        add_payment(post, payment, options)
+        commit('store', post, options[:customer_id])
+      end
+
+      def create_customer(options={})
+        post = {}
+        post[:name] = options[:name]
+        commit('customer', post)
+      end
```

Comment:
We typically don't introduce new methods to AM's public API. I can't say for certain that it hasn't happened before, but I believe it's discouraged. When a gateway requires additional setup, like creating a customer, the process is encapsulated in the public API . So instead of increasing the public footprint, would it be possible to make `create_customer` private or move the logic to `add_customer`?

@curiousepic - Any thoughts here? You've worked with AM longer than I have.

Reply:
I can make it private.